pytube/streams.py

```diff
@@ -271,6 +271,7 @@ def download(
                     bytes_remaining -= len(chunk)
                     # send to the on_progress callback.
                     self.on_progress(chunk, fh, bytes_remaining)
+            fh.close()
```

Comment:
You most likely do not need to call close() on the file handle if you already use a context manager (the `open(...) as fh` above).